benchmarks/bench_simul.c

```diff
@@ -386,7 +387,7 @@ run_ops(uint64_t ops, rng_t *rng, uint64_t *lat, void *get_buffer)
 
 			uint64_t size = min_size
 				+ (uint64_t)((double)(max_size - min_size + 1)
-				* rndlength(hash64(obj)));
+				* rndlength(hash64(obj ^ vsize_seed)));
```

Comment:
Is this really enough? obj is mostly zeroes, so xoring it with a constant will not generate much randomness.

Reply:
Yeah, but it won't degrade compared to what we already have.
 * obj is key id — mostly zeroes
 * vsize_seed is high quality randomness (seed or getentropy() → splitmix64 → xoshiro256**)

The result will then indeed go through just a single round of splitmix64 then get used for something that cares mostly about highest bits (in all three distributions we have).

So, you're right that what we have may have problems, but this particular commit won't make it any worse.